pandas/_libs/tslibs/_timestamp.pyx

```diff
@@ -0,0 +1,412 @@
+        if self.tzinfo is None:
+            if other.tzinfo is not None:
+                raise TypeError('Cannot compare tz-naive and tz-aware '
+                                'timestamps')
+        elif other.tzinfo is None:
+            raise TypeError('Cannot compare tz-naive and tz-aware timestamps')
+
+    cpdef datetime to_pydatetime(_Timestamp self, bint warn=True):
+        """
+        Convert a Timestamp object to a native Python datetime object.
+
+        If warn=True, issue a warning if nanoseconds is nonzero.
+        """
+        if self.nanosecond != 0 and warn:
+            warnings.warn("Discarding nonzero nanoseconds in conversion",
+                          UserWarning, stacklevel=2)
+
+        return datetime(self.year, self.month, self.day,
+                        self.hour, self.minute, self.second,
+                        self.microsecond, self.tzinfo)
+
+    cpdef to_datetime64(self):
+        """
+        Return a numpy.datetime64 object with 'ns' precision.
+        """
+        return np.datetime64(self.value, 'ns')
+
+    def to_numpy(self, dtype=None, copy=False):
+        """
+        Convert the Timestamp to a NumPy datetime64.
+
+        .. versionadded:: 0.25.0
+
+        This is an alias method for `Timestamp.to_datetime64()`. The dtype and
+        copy parameters are available here only for compatibility. Their values
+        will not affect the return value.
+
+        Returns
+        -------
+        numpy.datetime64
+
+        See Also
+        --------
+        DatetimeIndex.to_numpy : Similar method for DatetimeIndex.
+        """
+        return self.to_datetime64()
+
+    def __add__(self, other):
+        cdef:
+            int64_t other_int, nanos
+
+        if is_timedelta64_object(other):
+            other_int = other.astype('timedelta64[ns]').view('i8')
+            return self.__class__(self.value + other_int,
+                                  tz=self.tzinfo, freq=self.freq)
+
+        elif is_integer_object(other):
+            maybe_integer_op_deprecated(self)
+
+            if self is NaT:
+                # to be compat with Period
+                return NaT
+            elif self.freq is None:
+                raise ValueError("Cannot add integral value to Timestamp "
+                                 "without freq.")
+            return self.__class__((self.freq * other).apply(self),
+                                  freq=self.freq)
+
+        elif PyDelta_Check(other) or hasattr(other, 'delta'):
+            # delta --> offsets.Tick
+            # logic copied from delta_to_nanoseconds to prevent circular import
+            if hasattr(other, 'nanos'):
+                nanos = other.nanos
+            elif hasattr(other, 'delta'):
+                nanos = other.delta
+            elif PyDelta_Check(other):
+                nanos = (other.days * 24 * 60 * 60 * 1000000 +
+                         other.seconds * 1000000 +
+                         other.microseconds) * 1000
```

Comment:
Ugh, this comment got dropped again due to 'review'.  I've ensured this was the last missing comment I had.

As comment above implies, this logic was copied from `delta_to_nanoseconds`, has also been cleaned up of a few unnecessary `if`s.

`delta_to_nanoseconds` is a very independent function though, and if we want to move that to a separate file that would work too.